modules/imaging_uploader/php/imaging_uploader.class.inc

```diff
@@ -333,12 +358,15 @@ class Imaging_Uploader extends \NDB_Menu_Filter_Form
         //////////////.gz, .zip or .tgz////////////////////////////////////////
         ///////////////////////////////////////////////////////////////////////
         if (!$this->isCompressed($temp_file)) {
-            $errors[] = "The file $temp_file is not of type".
-                        " .tgz, .tar.gz or .zip. Please refresh this
-                        page to return to the Imaging Uploader module";
+            array_push($errors,
+                       "The file $temp_file is not of type".
+                       " .tgz, .tar.gz or .zip.");
         }
 
         if (count($errors) > 0) {
+            array_push($errors,
+                       "Please refresh this page to return to".
+                       " the Imaging Uploader module.");
```

Comment:
If you would like to have this presented to the front in a way that conforms more with React, I suggest throwing a `400` error and then exiting the script with a json encoding of `$errors`. This will allow you to access all the errors in the error clause of the jquery ajax call in the jsx file.

Reply:
e.g. 
`http_response_code(400);
header('Content-Type: application/json; charset=UTF-8);
exit(json_encorde(['errors'=>$errors]));`